mlflow/pyfunc/__init__.py

```diff
@@ -1377,7 +1369,15 @@ def _predict_row_batch(predict_fn, args):
                         " (Since the columns were passed unnamed they are expected to be in"
                         " the order specified by the schema).".format(len(names), names, len(args))
                     )
-            pdf = pandas.DataFrame(data={names[i]: x for i, x in enumerate(args)}, columns=names)
+            pdf = pandas.DataFrame(
+                data={
+                    names[i]: x
+                    if isinstance(x, pandas.Series)
+                    else x.apply(lambda row: row.to_dict(), axis=1)
```

Comment:
Can we add a comment here to explain this is for coerce a pandas DataFrame into a pandas Series for StructType support?

Reply:
Added!